### PR TITLE
support bf16

### PR DESCRIPTION
**Pull request by etemadiamd**
# What does this PR do?

Added bf16 support.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by etemadiamd** [comment]
@mfuntowicz @amyeroberts Started a new PR for bf16 support here!
CC: @AdrianAbeyta @amathews-amd

**Comment by LysandreJik** [comment]
WDYT @ydshieh ?

The `torch.cuda.is_bf16_supported` command was introduced in PyTorch 1.10. As I believe we dropped support for torch 1.9, wdyt of this change?

**Comment by ydshieh** [comment]
+1 for this change ❤️ .

File: src/transformers/utils/import_utils.py [hunk 1]
```diff
@@ -289,28 +289,7 @@ def is_torch_bf16_gpu_available():
...
-        if int(torch.version.cuda.split(".")[0]) < 11:
-            return False
-        if not hasattr(torch.cuda.amp, "autocast"):
-            return False
-    else:
-        return False
-
-    return True
-
+    return torch.cuda.is_bf16_supported()
```
Comment (ydshieh) [inline src/transformers/utils/import_utils.py:292]: We might need have `torch.cuda.is_available()` too here, as CI fails with `Found no NVIDIA driver on your system. Please check that you have an NVIDIA GPU `

Comment (etemadiamd) [reply src/transformers/utils/import_utils.py]: @ydshieh Thanks for the feedback. Added torch.cuda.is_available() .

**Comment by ydshieh** [comment]
Hi @etemadiamd I think your PR is on top of a commit on `main` that is old. Could you pull the latest changes to your local main and rebase your PR branch on top of `main`? This should fix several failing tests.

**Comment by ydshieh** [comment]
need a `make fixup` (or simply `make style` and `make quality`) to pass CI

**Comment by etemadiamd** [comment]
Hi @ydshieh, Thanks! Applied "make style and make quality". The "ci/circleci: check_code_quality" and "setup_and_quality" tests passed the CI checks. Now "run_tests" and "ci/circleci: examples_torch" tests failed with "Failed: Timeout >120.0s". Any idea to fix the failures?

**Comment by ydshieh** [comment]
Thanks for the fix! For the timeout one, we can ignore it.

**Review by amyeroberts — APPROVED** [review]
Thanks for this improvement!

**Comment by ydshieh** [comment]
Thanks again @etemadiamd !

**Pull request merged**